### PR TITLE
Continue Vehicle Update

### PR DESCRIPTION
**Pull request by duncanspumpkin**

### CODE REVIEW

File: src/openloco/objects/vehicle_object.h [hunk 1]
```diff
@@ -80,7 +128,13 @@ namespace openloco
...
         uint8_t pad_118;
         uint8_t startsnd_type; // 0x119
-        uint8_t pad_11A[0x15A - 0x11A];
+        union
+        {
+            vehicle_object_sound_1 type_1;
+            vehicle_object_sound_2 type_2;
+            vehicle_object_sound_3 type_3;
+        } sound;
+        uint8_t pad_135[0x15A - 135];
```
Comment (duncanspumpkin) [inline src/openloco/objects/vehicle_object.h]: 0x135?

File: src/openloco/map/tile.h [hunk 2]
```diff
@@ -194,8 +206,8 @@ namespace openloco::map
         uint8_t _7;
 
     public:
-        bool has_40() const { return (_type & 0x40) != 0; }
-        bool has_80() const { return (_type & 0x80) != 0; }
+        bool has_signal() const { return (_type & 0x40) != 0; }
```
Comment (duncanspumpkin) [inline src/openloco/map/tile.h]: Confirm this!

File: src/openloco/window.cpp [hunk 3]
```diff
@@ -107,14 +107,14 @@ namespace openloco::ui
 
             if (config->viewport_target_sprite != 0xFFFF)
             {
-                auto thing = thingmgr::get<thing_base>(config->viewport_target_sprite);
+                auto _thing = thingmgr::get<thing>(config->viewport_target_sprite);
```
Comment (marijnvdwerf) [inline src/openloco/window.cpp]: Good example of why I created #193

File: src/openloco/things/vehicle.cpp [hunk 4]
```diff
@@ -1,36 +1,96 @@
...
+loco_global<uint8_t, 0x00525FAE> vehicle_var_525FAE;       // boolean
+loco_global<uint8_t[128000], 0x987C5C> vehicle_var_987C5C; // Size tbc
+loco_global<uint32_t[7], 0x004FE070> vehicle_var_4FE070;   // Size tbc
+loco_global<uint8_t[7], 0x004FE088> vehicle_var_4FE088;    // Size tbc
+loco_global<station_id_t, 0x01135FAE> vehicle_var_1135FAE;
+loco_global<uint8_t, 0x0113607D> vehicle_var_113607D;   // bool
+loco_global<uint16_t*, 0x01135EE6> vehicle_var_1135EE6; // vehicle_var_11360A0 related?
+loco_global<uint16_t, 0x01136458> vehicle_var_1136458;
+#pragma pack(push, 1)
+struct unk_4F73D8
```
Comment (marijnvdwerf) [inline src/openloco/things/vehicle.cpp]: Appears to be the same structure as `rct_preview_track`

**Pull request closed without merging**